### PR TITLE
Hand out temporary loopback addresses from a larger pool

### PR DESCRIPTION
**Pull request by raboof**
Might have prevented https://github.com/akka/akka/issues/23528

### CODE REVIEW

File: akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala [hunk 1]
```diff
@@ -25,21 +26,28 @@ object SocketUtil {
...
   }
 
-  def temporaryServerAddress(address: String = "127.0.0.1", udp: Boolean = false): InetSocketAddress =
+  val RANDOM_LOOPBACK_ADDRESS = "RANDOM_LOOPBACK_ADDRESS"
+
+  def temporaryServerAddress(address: String = RANDOM_LOOPBACK_ADDRESS, udp: Boolean = false): InetSocketAddress =
     temporaryServerAddresses(1, address, udp).head
 
-  def temporaryServerAddresses(numberOfAddresses: Int, hostname: String = "127.0.0.1", udp: Boolean = false): immutable.IndexedSeq[InetSocketAddress] = {
+  def temporaryServerAddresses(numberOfAddresses: Int, hostname: String = RANDOM_LOOPBACK_ADDRESS, udp: Boolean = false): immutable.IndexedSeq[InetSocketAddress] = {
```
Comment (patriknw) [inline akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala]: interesting
search for "temporaryServerAddress", we sometimes specify the `hostname` and that should be removed

Comment (raboof) [inline akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala]: Good one - I noticed some tests disregard the host part and assume 'localhost', perhaps we should create a separate `temporaryLocalPort` for those use cases?

**Comment by raboof** [comment]
Hmm I just noticed this code somewhere:
```
        try SocketUtil.temporaryServerAddress(address = "127.0.1.1", udp = proto == "udp")
        catch {
          case e: java.net.BindException ⇒
            info(s"skipping test due to [${e.getMessage}], you probably have to use `ifconfig lo0 alias 127.0.1.1`")
            pending
            null
        }
```

This suggests `127.0.1.1/8` isn't always a loopback address by default. Seems to work on my machine and Jenkins though

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala (hunk 1 above)
Comment (ktoso) [reply akka-testkit/src/main/scala/akka/testkit/SocketUtil.scala]: No strong opinion on that, good to merge as is I think

**Review by ktoso — APPROVED** [review]

**Pull request merged**